components/clp-rust-utils/src/s3/client.rs

```diff
@@ -6,10 +7,16 @@ use secrecy::{ExposeSecret, SecretString};
 
 /// Creates a new S3 client.
 ///
+/// # Notes
+///
+/// * The client is configured using the latest AWS SDK behavior version.
+/// * The client enforces path-style addressing.
+///
 /// # Returns
+///
 /// A newly created S3 client.
 #[must_use]
-pub fn create_new_client(
+pub async fn create_new_client(
```

Comment:
changed as `async` which is an internal breaking change -

i confirmed that there's no usage of the function at the moment and therefore there's no caller to be updated